File: week08/2/Prob5.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        left = 0
        right = len(nums) - 1

        while left < right:
            mid = (left + right) // 2
            if nums[mid] > nums[right]:
                # minimum must be in the right part
                left = mid + 1
            elif nums[mid] < nums[right]:
                # minimum must be in the left part
                right = mid
            else:
                right -= 1
        return nums[left]

File: week08/5/prob5.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int findMin(vector<int>& nums) {
        int minimum = nums.front();
        for (int n : nums) {
            if (n < minimum) {
                minimum = n;
                break;
            }
        }
        return minimum;
    }
};
